pkg/renamepvc/renamepvc_test.go

```diff
@@ -0,0 +1,247 @@
+					Name: "test",
+					VolumeSource: corev1.VolumeSource{
+						PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+							ClaimName: "test3-old", ReadOnly: false,
+						},
+					},
+				}},
+			},
+		}
+
+		o, err := initTestSetup(ctx,
+			"test3-old", "test3",
+			"test3-new", "test3",
+			&pod,
+		)
+		if err != nil {
+			t.Errorf("testsetup failed - got error %q", err)
+		}
+
+		err = o.run(ctx)
+		if !errors.Is(err, ErrVolumeMounted) {
+			t.Errorf("expect %q error - but got %q", ErrVolumeMounted, err)
+		}
+	})
+
+	t.Run("Test rename-pvc with running pod - fail", func(t *testing.T) {
+		ctx := context.Background()
+
+		pvc := corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{Name: "test4-new", Namespace: "test4"},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				VolumeName: "test",
+			},
+		}
+
+		o, err := initTestSetup(ctx,
+			"test4-old", "test4",
+			"test4-new", "test4",
+			&pvc,
+		)
+		if err != nil {
+			t.Errorf("testsetup failed - got error %q", err)
+		}
+		ll, _ := o.k8sClient.CoreV1().PersistentVolumeClaims("default").List(ctx, metav1.ListOptions{})
+		_ = ll
+		err = o.run(ctx)
+		if !k8sErrors.IsAlreadyExists(err) {
+			t.Errorf("expect already exists error - but got %q", err)
+		}
+	})
+}
+
+func TestConfirmCheck(t *testing.T) {
+	t.Run("Test confirmCheck with y", func(t *testing.T) {
+		streams, in, _, _ := genericclioptions.NewTestIOStreams()
+		o := renamePVCOptions{streams: streams}
+		in.WriteString("y\n")
+		if err := o.confirmCheck(); err != nil {
+			t.Errorf("confirmCheck - got error %q", err)
+		}
+	})
+	t.Run("Test confirmCheck with skip flag", func(t *testing.T) {
+		streams, _, _, _ := genericclioptions.NewTestIOStreams()
+		o := renamePVCOptions{streams: streams, confirm: true}
+		if err := o.confirmCheck(); err != nil {
+			t.Errorf("confirmCheck - got error %q", err)
+		}
+	})
+	t.Run("Test confirmCheck with n", func(t *testing.T) {
+		streams, in, _, _ := genericclioptions.NewTestIOStreams()
+		o := renamePVCOptions{streams: streams}
+		in.WriteString("n\n")
+		if err := o.confirmCheck(); !errors.Is(err, ErrAcceptationNotYes) {
+			t.Errorf("expect %q - got error %q", ErrAcceptationNotYes, err)
+		}
+	})
+	t.Run("Test confirmCheck with unknown", func(t *testing.T) {
+		streams, in, _, _ := genericclioptions.NewTestIOStreams()
+		o := renamePVCOptions{streams: streams}
+		in.WriteString("unknown\n")
+		if err := o.confirmCheck(); !errors.Is(err, ErrAcceptationUnknown) {
+			t.Errorf("expect %q - got error %q", ErrAcceptationUnknown, err)
+		}
+	})
+}
+
+// initTestSetup returns renamePVCOptions initialized with a kubernetes fake client and created oldPvc for test
+func initTestSetup(
+	ctx context.Context,
+	oldName, sourceNamespace string,
+	newName, targetNamespace string,
+	extraObjects ...runtime.Object,
+) (renamePVCOptions, error) {
+	// init client
+	streams, _, _, _ := genericclioptions.NewTestIOStreams() //nolint:dogsled // not needed for tests
```

Comment:
We can just add this to the overall linting config to exclude dogsled from test files